monai/networks/blocks/upsample.py

```diff
@@ -71,3 +71,64 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             x: Tensor in shape (batch, channel, spatial_1[, spatial_2, ...).
         """
         return torch.as_tensor(self.upsample(x))
+
+
+class SubpixelUpsample(nn.Module):
+    """
+    Upsample via using a subpixel CNN. This module supports 1D, 2D and 3D input images.
+    The module is consisted with two parts. First of all, a convolutional layer is employed
+    to increase the number of channels into: ``in_channels * (scale_factor ** spatial_dims)``.
+    Secondly, a pixel shuffle manipulation is utilized to aggregates the feature maps from
+    low resolution space and build the super resolution space.
+    The first part of the module is not fixed, a sequential layers can be used to replace a
+    single layer. 
+    The idea comes from:
+    https://arxiv.org/abs/1609.05158
+    The pixel shuffle mechanism refers to:
+    https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/PixelShuffle.cpp
+    and:
+    https://github.com/pytorch/pytorch/pull/6340/files
+    """
+
+    def __init__(self, spatial_dims: int, in_channels: int, scale_factor: int = 2,) -> None:
+        """
+        Args:
+            spatial_dims: number of spatial dimensions of the input image.
+            in_channels: number of channels of the input image.
+            scale_factor: multiplier for spatial size. Defaults to 2.
+        """
+        super().__init__()
+        conv_out_channels = in_channels * (scale_factor ** spatial_dims)
+        self.spatial_dims = spatial_dims
+        self.scale_factor = scale_factor
```

Comment:
need some validation such as `scale_factor > 1`?

Reply:
That makes sense, let me add it.